### DISCUSSION TITLE
VS code server not listing kernel created from conda virtual environment

### DISCUSSION THREAD

**Discussion by Ritesh313** [microsoft/vscode-jupyter#9438] [General] [answered]
I'm working on a linux based server. I ssh to the server and able to access all the files.
I have a conda environment which is not in the usual .conda folder, because of limites storage space it needed to be created in a different location. 
Python           : 3.9.7
IPython          : 8.1.1
ipykernel        : 6.9.2
ipywidgets       : 7.7.0
jupyter_client   : 7.1.0
jupyter_core     : 4.9.1
jupyter_server   : not installed
jupyterlab       : not installed
nbclient         : 0.5.13
nbconvert        : not installed
nbformat         : 5.2.0
notebook         : 6.4.8

Here are the commands I run after ssh to the server:
1. `conda activate /path_to_my_conda_environment/`
2. I press `ctrl+shift+p` to select python interpreter which is in `/path_to_my_conda_environment/bin/python`, I am able to run my python script without any issues using this environment.
![image](https://user-images.githubusercontent.com/36135489/158874468-5543652e-fb91-4fa1-a341-78ed52e79751.png)

3. The available jupyter kernels using this command `jupyter kernelspec list`: 
![image](https://user-images.githubusercontent.com/36135489/158876553-2e5dcc00-c8eb-4f35-83e7-b5df50e9cab8.png)

4. I have activated this virtual conda environment as a Jupyter kernel, I have confirmed that it shows up as a jupyter kernel on the jupyter hub on the server, but when I run a jupyter notebook on vs code (connected to the same server by ssh), I don't see this kernel. The image below shows the kernel blue_jupy on the jupyter hub.
![Screenshot (1)](https://user-images.githubusercontent.com/36135489/158875315-e6d48740-5c8b-48ac-9091-017d6e3f4cbe.png)

6. I have confirmed that if I create a conda environment without any prefix (which creates it in .conda folder in the home directory), I am able to see it as a jupyter kernel in vs code but not the environment which has a prefix.
![Screenshot (2)](https://user-images.githubusercontent.com/36135489/158876137-31aff4ab-1304-40d7-9b8f-aeadf8e3f36e.png)

Can anyone point me to a discussion or help me find what's the issue here? Please let me know if I should add any other information here for clarity.

**Comment by rchiodo** [comment, marked as answer]
@Ritesh313 without the environment being in the conda env folder, we cannot determine the environment name. Unfortunately you'll have to use the path to distinguish it (as you did for selecting the python interpreter).
